api/src/main/java/org/openmrs/module/reporting/definition/library/DefinitionLibraryFactory.java

```diff
@@ -0,0 +1,16 @@
+package org.openmrs.module.reporting.definition.library;
+
+import java.util.Collection;
+
+/**
+ * Implementing a bean of this interface allows you to dynamically create zero or more {@link DefinitionLibrary}s.
+ */
+public interface DefinitionLibraryFactory {
+
+	/**
+	 * An implementation of this interface can count on this method being invoked once by the Reporting module's activator.
+	 * @return 0 or more definition libraries you want to register
+	 */
+	Collection<DefinitionLibrary<?>> getDefinitionLibraries();
+	
+}
```

Comment:
@djazayeri is this Factory pattern really needed?  Do you see modules looking to add onto this functionality wanting to add their own factories?  If this is actually needed, shouldn't the pre-existing DocumentedDefinition-based definition libraries also get exposed via a Factory implementation?